src-browserify/api/sql.js

```diff
@@ -1,686 +1,697 @@
 var _ = require('underscore');
-var _Promise = require('_Promise');
 var Mustache = require('mustache');
 
-// TODO: jQuery.ajax OR reqwest is expected to be located at global namespace when an SQL object is created
-function SQL(options) {
-  if(cartodb === this || window === this) {
-    return new SQL(options);
-  }
-  if(!options.user) {
-    throw new Error("user should be provided");
-  }
-  var loc = new String(window.location.protocol);
-  loc = loc.slice(0, loc.length - 1);
-  if(loc == 'file') {
-    loc = 'https';
-  }
-
-  this.ajax = options.ajax || (typeof(jQuery) !== 'undefined' ? jQuery.ajax: reqwest);
-  if(!this.ajax) {
-    throw new Error("jQuery or reqwest should be loaded");
-  }
-
-  this.options = _.defaults(options, {
-    version: 'v2',
-    protocol: loc,
-    jsonp: typeof(jQuery) !== 'undefined' ? !jQuery.support.cors: false
-  })
-
-  if (!this.options.sql_api_template) {
-    var opts = this.options;
-    var template = null;
-    if(opts && opts.completeDomain) {
-      template = opts.completeDomain;
-    } else {
-      var host = opts.host || 'cartodb.com';
-      var protocol = opts.protocol || 'https';
-      template = protocol + '://{user}.' + host;
+// NOTE this does not return a SQL directly, but a wrapper, to inject the dependencies
+// e.g. var SQL = require('./sql')({ jQuery: $ /* or reqwest: reqwest */ });
+// @param {Object} _Promise
+// @param {Object} opts hash with keys:
+//   jQuery: {Object} (Optional) has precedence over reqwest if both are present
+//   reqwest: {Object} (Optional)
+module.exports = function(_Promise, opts) {
```

Comment:
@alonsogarciapablo thx to @rochoa came up with a better solution :tada: the problem remains ofc, but with the alternative approach we avoid the annoying wrapper/setupXX I've used so far. 

I'll explain here with `src/core/model.js` as example, which is used in all 4 bundles (removed real code for brevity):

Instead of require dependencies right away, set through through a _proxy_ at entry point (i.e. the bundle):
```js
// standard.js
var $ = require('jquery');
require('../proxies/ajax').set($.ajax);
…
```
```js
// no-jquery.js
var reqwest = require('reqwest');
require('../proxies/ajax').set(reqwest.compat);
…
```
```js
// core.js
var reqwest = require('reqwest');
require('../proxies/ajax').set(reqwest.compat);

var cdb = {};
cdb.SQL = require('./api/sql');
…
```

Using code requires ajax thorugh the proxy object:
```js
// api/sql.js
var ajax = require('../proxies/ajax').get(); // get actual ajax function to use

function SQL() {
  this.ajax = options.ajax || ajax;
  // old code, for comparison:
  // this.ajax = options.ajax || (typeof(jQuery) !== 'undefined' ? jQuery.ajax: reqwest);
  ...

module.exports = SQL;
```

So the dependency is wired up at runtime, with the benefit of having the code declarative (through the requires). Sounds OK?